pytmc/xml_collector.py

```diff
@@ -874,72 +736,122 @@ def valid(self):
                                   for key in self._required_fields)
         return has_required_keys and has_required_fields
 
-    def render_record(self):
+    @property
+    def records(self):
+        """Generated :class:`.EPICSRecord` objects"""
+        return NotImplemented
+
+    def render_records(self):
         """
         Returns
         -------
         string
-            Jinja rendered entry for this BaseRecordPackage
+            Jinja rendered entry for the RecordPackage
         """
-        simple_dict = self.cfg_as_dict()
         if not self.valid:
-            logger.error('Unable to render record: %s', simple_dict)
+            logger.error('Unable to render record: %s', self)
             return
+        return '\n'.join([record.render_template()
+                          for record in self.records])
+
+    @classmethod
+    def from_chain(cls, *args, chain, **kwargs):
+        """Select the proper subclass of ``TwincatRecordPackage`` from chain"""
+        last = chain.last
+        if last.is_array:
+            spec = WaveformRecordPackage
+        elif last.tc_type in {'STRING'}:
+            spec = StringRecordPackage
+        else:
+            spec = data_types[last.tc_type]
+        # Create a RecordPackage from the chain
+        return spec(*args, chain=chain, **kwargs)
 
-        return self.record_template.render(**simple_dict)
 
-    @staticmethod
-    def generate_files(records):
+class TwincatTypeRecordPackage(RecordPackage):
+    """
+    The common parent for all RecordPackages for basic Twincat types
+
+    This main purpose of this class is to handle the parsing of the pragma
+    chains that will be shared among all variable types. This includes setting
+    the appropriate asyn port together and handling the "io" directionality. If
+    you have a :class:`.TmcChain` but are not certain which class is
+    appropriate use the :meth:`.from_chain` class constructor and the correct
+    subclass will be chosen based on the given variable information
+
+    In order to subclass:
```

Comment:
```suggestion
    In order to subclass::
```

I think then the next block will have to be indented (check the sphinx output).
Also, these classes need to be added to the autoclass lists in the docs.

Reply:
I don't think the indent did what you expected, added classes to docs.